CMakeLists.txt

```diff
@@ -1,198 +1,247 @@
-  target_link_libraries(rvt_test
-    ${PROJECT_NAME}
-    ${catkin_LIBRARIES}
-    ${Boost_LIBRARIES}
-  )
-
-endif()
-
-## Test for correct C++ source code
-roslint_cpp()
+ament_target_dependencies(${PROJECT_NAME}_demo
+  rclcpp
+)
 
 #############
 ## Install ##
 #############
 
+ament_export_libraries(${PROJECT_NAME})
+ament_export_libraries(${PROJECT_NAME}_remote_control)
+# ament_export_libraries(${PROJECT_NAME}_gui)
+ament_export_libraries(${PROJECT_NAME}_imarker_simple)
+
 # Install libraries
 install(
   TARGETS
     ${PROJECT_NAME}
-    ${PROJECT_NAME}_gui
-    ${PROJECT_NAME}_imarker_simple
     ${PROJECT_NAME}_remote_control
-  LIBRARY DESTINATION
-    ${CATKIN_PACKAGE_LIB_DESTINATION}
-  ARCHIVE DESTINATION
-    ${CATKIN_PACKAGE_LIB_DESTINATION}
-  RUNTIME DESTINATION
-    ${CATKIN_GLOBAL_BIN_DESTINATION}
+    # ${PROJECT_NAME}_gui
+    ${PROJECT_NAME}_imarker_simple
+  LIBRARY DESTINATION lib
+  ARCHIVE DESTINATION lib
+  RUNTIME DESTINATION bin
 )
 
-# Install header files
-install(DIRECTORY include/${PROJECT_NAME}/   DESTINATION ${CATKIN_PACKAGE_INCLUDE_DESTINATION})
+# Install executables
+install(
+  TARGETS
+    ${PROJECT_NAME}_demo
+    ${PROJECT_NAME}_imarker_simple_demo
+  LIBRARY DESTINATION lib
+  ARCHIVE DESTINATION lib
+  RUNTIME DESTINATION lib/${PROJECT_NAME}
+)
 
 # Install shared resources
-install(DIRECTORY launch    DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION})
-install(DIRECTORY resources DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION})
+install(DIRECTORY launch    DESTINATION share/${PROJECT_NAME})
+# install(DIRECTORY meshes    DESTINATION share/${PROJECT_NAME})
 
 # Install xml plugin config
-install(FILES
-  plugin_description.xml
-  DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION}
-)
+# install(FILES
+#   plugin_description.xml
+#   DESTINATION share/${PROJECT_NAME}
+# )
 
-# Install executables
-install(TARGETS ${PROJECT_NAME}_demo ${PROJECT_NAME}_imarker_simple_demo
-  LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-  ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-  RUNTIME DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
+# Install header files
+install(
+  DIRECTORY include/
+  DESTINATION include)
+
+##########
+## TEST ##
+##########
+
+if (BUILD_TESTING)
+  # # Lint
+  # find_package(ament_lint_auto REQUIRED)
+  # ament_lint_auto_find_test_dependencies()
+
+  # # Run tests
+  # find_package(ament_cmake_gtest REQUIRED)
+  # ament_add_gtest(${PROJECT_NAME}_rvt_test tests/rvt_test.cpp)
+  # ament_target_dependencies(${PROJECT_NAME}_rvt_test
+  #   rclcpp
+  # )
+  # target_link_libraries(${PROJECT_NAME}_rvt_test
+  #   ${PROJECT_NAME}
+  # )
+endif()
+
+
+ament_export_include_directories(include)
+ament_export_dependencies(
+  # eigen_conversions
+  # tf_conversions
+  # graph_msgs
+  geometry_msgs
+  shape_msgs
+  roscpp
+  sensor_msgs
+  std_msgs
+  trajectory_msgs
+  visualization_msgs
+  eigen3_cmake_module
+  Eigen3
```

Comment:
Your find_package list seemed longer, might be missing some stuff here

find_package(ament_cmake REQUIRED)
find_package(sensor_msgs REQUIRED)
find_package(geometry_msgs REQUIRED)
find_package(shape_msgs REQUIRED)
find_package(visualization_msgs REQUIRED)
find_package(std_msgs REQUIRED)
find_package(trajectory_msgs REQUIRED)
find_package(eigen3_cmake_module REQUIRED)
find_package(eigen_stl_containers REQUIRED)
find_package(Eigen3 REQUIRED)	find_package(Eigen3 REQUIRED)
find_package(Boost REQUIRED thread system)	find_package(Boost REQUIRED thread system)
find_package(PkgConfig REQUIRED)	find_package(PkgConfig REQUIRED)
pkg_check_modules(OGRE_OV OGRE)	find_package(tf2 REQUIRED)
find_package(tf2_eigen REQUIRED)
find_package(tf2_geometry_msgs REQUIRED)
find_package(interactive_markers REQUIRED)
find_package(Qt5 REQUIRED COMPONENTS Widgets)